GeoLib/Polygon.cpp

```diff
@@ -38,6 +38,25 @@ Polygon::Polygon(Polygon const& other)
 	}
 }
 
+Polygon& Polygon::operator=(Polygon const& other)
+{
+	if (other._ply_pnts != _ply_pnts) {
+		ERR(
+		    "Polygon::operator=: polygon other and this polygon are not based "
+		    "on the same point vector. Aborting");
```

Comment:
I just saw that `const std::vector<Point*> &_ply_pnts;` is a reference member of `Polyline`. Therefore it cannot be reset by the assignment operator.
Different thing: However, it would probably be cheaper to compare the addresses and not the whole vectors. Furthermore the check `other == *this` should probably be done first.